src/main/scala/chisel3/util/Decoupled.scala

```diff
@@ -265,14 +281,23 @@ class Queue[T <: Data](val gen: T,
   }
 
   val ptr_diff = enq_ptr.value - deq_ptr.value
+  val flush = Wire(Bool())
+
+  if (hasFlush) { 
+    flush := io.flush.get //to mitigate returning None from io.flush.get in logic below
+  } 
+  else { 
+    flush := false.B 
+  }
+
   if (isPow2(entries)) {
-    io.count := Mux(maybe_full && ptr_match, entries.U, 0.U) | ptr_diff
+    io.count := Mux(maybe_full && !flush && ptr_match, entries.U, 0.U) | ptr_diff
   } else {
     io.count := Mux(ptr_match,
-                    Mux(maybe_full,
-                      entries.asUInt, 0.U),
+                    Mux(maybe_full && !flush,
+                      entries.asUInt(), 0.U),
```

Comment:
```suggestion
                      entries.asUInt, 0.U),
```
Presumably Scala 2.13 is warning you about this, but I would rather fix the definition of `.asUInt` than add these parentheses.